### ISSUE TITLE
Reimplement Addons items

### ISSUE THREAD

**Issue by TeamSpen210** [TeamSpen210/BEE2-items#95]
(Moved from BenVlodgi/BEE2.4#98)
* [ ] Retractable Pedestal button
* [ ] Worldportal Door
* [ ] Diagonal Laser and High Energy Pellet launcher
* [ ] Quarter wall
* [ ] Dropper version of PortalCombat's Sphere Cannon
* [X] Antline Router
* [X] Antline Magnet
* [ ] Angled Block
* [X] ~~Music~~ (Merged into bee2 itself)
* [ ] ~~Laser Music~~ (Add automatically via compiler)
* [ ] Conveyor Belt
* [ ] Large Faith Plate
* [ ] Global Cube Dropper
* [ ] Paint Cleaner
* [ ] Double Grated Panel
* [ ] Futbol
* [ ] Ditch
* [ ] Caged Turret
* [ ] Caged Reflection Cube
* [ ] Wheatley Monitor
* [X] ~~Portalable Stair~~ (Merged into default)
* [ ] Portalable Piston Platform
* [ ] Turret Dropper
* [ ] Toggleable Faith Plate (Merge into default)
* [X] ~~Static Panel~~ (Merged into default)
* [ ] Static Piston Platform
* [ ] Half/Half Wall
* [X] Coop Checkpoint
* [ ] Recessed Fizzlers
* [ ] Crusher Plates
* [ ] Sideways Light Bridges
* [ ] Shredders
* [X] ~~Static Glass Panels~~ (Merged into default)
* [ ] Half Panels
* [ ] BumbleBalls
* [ ] Catapult Cubes
* [ ] Bomb Droppers
* [ ] Rotating Faith Plates
* [ ] Glass With Hole
* [ ] Glass Doors
* [ ] Slow Crushers
* [ ] Piston Relays
* [X] SR Latches
* [ ] Angled Light Bridge
* [X] Separation Coop Checkpoint
* [ ] Cube Transmuters
* [ ] Ceiling Lights
* [ ] Autoportal Pistons
* [ ] Timeout Portals
* [ ] Clamped Cubes
* [ ] Global Cube Triggers
* [ ] Override
* [ ] Neurotoxin
* [X] Chamberlock
* [ ] Retractable Floor Button

**Comment by ellingtonisland** [comment]
- see if you can find a way to make the recessed fizzler easily sizeable like normal fizzlers are so i don't have to place a bunch of 1x1 voxel wide fizzlers across a floor.
- add a 2x2 voxel centered funnel, faith plate, and accomp faith plate target if you can
- change the logic of the catwalk so the sections can also be extended out from the wall, floor, or ceiling like the piston platform arm can.
- who will you put on the wheatley monitor for BTS style?

**Comment by TeamSpen210** [comment]
I should probably make a ¾ block item instead, that way you can use regular fizzler items as well. 

What I'm planning on doing with monitors is make them dependant on both the style and voice lines. Different styles will use different monitor designs (the Wheatley screen, computer monitors in BTS, old screens in Old Aperture or portraits). Who's displayed in the screen depends on the voice line. For the Announcer the screen should display static, or one of the computer readout videos from the Coop campaign.

I can't use the piston platform logic, but I might be able to make it use the floor button item type to extend it from the wall. The problem with that is it'll require weirdly long supports in some cases.

**Comment by TheReviewers** [comment]
@TeamSpen210 Can You Add An Fizzler That Can Go With The Thin Wall?

**Comment by Konclan** [comment]
Kinda like the paint fizzlers in aperture tag?

**Comment by TheReviewers** [comment]
Yes

**Comment by TeamSpen210** [comment]
I can't really add modifications to fizzlers, it's hard coded the way that the resizable ones work..

**Comment by TheReviewers** [comment]
@TeamSpen210 How To Add Fizzlers To Thin Walls Then?

**Comment by LautaroL20** [comment]
@TheReviewers - I think you mean something like this, check this link: 

"Standing Placeable Fizzler / LaserField" https://github.com/TeamSpen210/BEE2-items/issues/257

**Comment by Konclan** [comment]
This list has not been updated yet..

**Comment by xDadiKx** [comment]
The Retractable Pedestal button is in BEE2 now. You can tick it.

**Comment by Luke18033** [comment]
@TeamSpen210 Are you also going to re-implement the items from the standard BEE2.2.5 (without addons)?

**Comment by TeamSpen210** [comment]
Yes

**Comment by Luke18033** [comment]
@TeamSpen210 Addons Items I would like to see next:
- Toggleable Faith Plate (Merge into default)
- Neurotoxin
- Retractable Floor Button
- Rexaura Pellet Detector Field - make them resizeable like normal fizzlers.
- Slow Crusher
- Crusher Plate
- Shredder (Spinny Blade Wall)
- Caged Turret

**Comment by Luke18033** [comment]
@TeamSpen210 Crusher plates should be generated automatically where crushers hit a wall. Also, for the Skybox item, Sphere Interior should generate an enterable opening to the sphere outside. Use worldportals for this.

**Comment by LautaroL20** [comment]
I don´t see the "Wall Feature" item anywhere on that list....

**Comment by Luke18033** [comment]
@LautaroL20 I think Wall Feature was in the base BEE2, not the addons. I'm sure it will still be implemented though.

**Comment by LautaroL20** [comment]
No, the "Wall Feature" is part of the TS Addons Pack. Here is a screen of the original base Bee2.2.5 Items, and this item wasn´t there:

![bee](https://cloud.githubusercontent.com/assets/12244376/21151891/f3a47812-c142-11e6-8677-35ae1b6232d4.png)

**Comment by TheDarkBomber** [comment]
What's the item that looks like a wall?

**Comment by LautaroL20** [comment]
That one is the Felix Griffin´s "P1 Wall Item" wich is totally different from the "Wall Feature" created by TeamSpen210.

**Comment by TeamSpen210** [comment]
Wall features would be replaced by BenVlodgi/BEE2.4#414.

**Comment by Luke18033** [comment]
@TeamSpen210 What about the crusher plates? Also, why is cube cleaner checked, it hasn't been added yet.

**Comment by LautaroL20** [comment]
Cube cleaner is already in our dev version, but it causes some failure to compiles.

**Comment by TeamSpen210** [comment]
Crusher plates would be a 'special' decoration, IE one that always tries to be placed under the crusher.

**Comment by Luke18033** [comment]
But wouldn't that mean it would only be generated sometimes, unless the amount slider is set to max? Or would it always generate?

**Comment by TeamSpen210** [comment]
I'd make it always generate, ignoring amounts and whatnot - it only wouldn't if there's no surface, or you put a blacklist item in that space.

**Comment by Luke18033** [comment]
@TeamSpen210 Couldn't you just use the antline router or something for the blacklist item?

**Comment by Luke18033** [comment]
And also, how about just have Start Enabled and Start Reversed for the portalable piston platform? Like the static version originally had.

**Comment by Luke18033** [comment]
Item improvements:
- [ ] Generate Crusher Plates automatically below crushers (via BEEmod/BEE2.4#414)
- [ ] Add Start Enabled and Start Reversed to Piston Platform. This would allow adding a solid panel (Start Enabled) and making that panel portalable (Start Reversed), like the static platform in the original addons.
- [ ] Make Catapult Cubes, Bumbleballs, Bombs, and Sphere Cannons placeable like normal cubes (dropper-less and with dropper) and add Cube Type. Additionally, add a stylevar to disable Sphere Cannon help text.
- [ ] Add a non-static half panel.
- [ ] Make the turret dropper place turrets with a claw arm, and allow them to be placed more than 2 blocks down.

**Comment by Luke18033** [comment]
@TeamSpen210 - What addon items will be added in Pr25? Physics Repulsion Field is already in, but do you have anything else planned? I'd like the Portalable Piston Platform (hopefully merged into default, see my previous comment), I need it for one of my maps.

**Comment by TeamSpen210** [comment]
I should make that, but currently I'm working on the split Glass Door panel item.

**Comment by TeamSpen210** [comment]
For Half Grate Panels, they'll be a Track Platform item class.

**Comment by libbybapa** [comment]
I really want the 'Turret Dropper' for a specific map idea.  Just saying.  :-)

**Comment by TheDarkBomber** [comment]
There should be a itemvar of the Cube Transmuter for BP cubes. The itemvar should be a Start Reversed option.

**Comment by Luke18033** [comment]
That would be a good idea. It wouldn't work with the Activation Cube though, since that has outputs but other cubes don't.

**Comment by TheDarkBomber** [comment]
Yes.

**Comment by Endermage77** [comment]
Is there a list like this that is ordered by priority rather than alphabet?

**Comment by Luke18033** [comment]
I added an additional suggestion to my improvements comment.

**Comment by Byzarru** [comment]
@Endermage77 I'd say they are all pretty much the same priority.

**Comment by Byzarru** [comment]
My two notes:
* Angled Panels and Partial Blocks should be used in conjunction with test elements to create their angled variants.
* Shredders and Bumbleballs are listed in the BEESigns.

**Comment by ZombiePwner1** [comment]
2 of my favourite items from bee2.5.5 i want reimplemented into bee2.4 as quickly as possible is the caged reflection cube and turret.

**Comment by libbybapa** [comment]
Are those different from just building a cage around a turret or reflective cube?

**Comment by Endermage77** [comment]
The caged turret appears in a challenge map for Portal 1: it's a very small grating cage just around the turret that makes the turret completely and utterly immovable, and thus a much more dangerous target.
Basically, the caged variant of the item is locked in place but otherwise unmodified.

**Comment by ZombiePwner1** [comment]
@Endermage77 yes you are correct with the caged turret however it can still be killed by lasers.

**Comment by libbybapa** [comment]
Ok.  I have a foggy memory of the caged turret in P1.  It's been a while...  How is a caged reflective cube functionally different from putting a reflective cube inside a glass box?  Just curious.

**Comment by TeamSpen210** [comment]
That'd be the same, except the box is just the right size, so it's a bit prettier and compact.

**Comment by libbybapa** [comment]
Ok.  Thanks for the explanation.  I still want a turret dropper most.  :)

**Comment by Endermage77** [comment]
Yeah, I agree. Preferably one without the fixed-distance landing plate.

**Comment by TeamSpen210** [comment]
That's definitely going away.

**Comment by Luke18033** [comment]
#4

**Comment by Luke18033** [comment]
#4

**Comment by Luke18033** [comment]
Why was the landing plate there anyway? Couldn't you just have used clips to prevent the turret from falling over?

**Comment by Luke18033** [comment]
Also, you should make it so that the Catapult Cube and Bumbleball can be colorised. Maybe also coloriseable bombs, but I'm not sure how useful that would be.

**Comment by TeamSpen210** [comment]
Because the box is also physics object, I needed to ensure I knew where it would land sp I could spawn in the real turret. The new version will use a claw arm, so it won't be an issue - it can extend forever if needed.

**Comment by Luke18033** [comment]
If it had clips then the turret wouldn't have needed to be boxed in the first place. Actually, clips would have needed special compiler logic to ensure they don't go outside the map or into a room below, and tspenAddons couldn't use the compiler like that.

**Comment by TeamSpen210** [comment]
It does have clips for the box, the problem is that it's a bit unrealistic for the turret to drop that far and not fall over.

**Comment by Luke18033** [comment]
Which is why a claw arm makes more sense.

**Comment by Luke18033** [comment]
The only problem with a claw arm is that the piston models (which I'm assuming you'd use) are only designed to extend 512 units at maximum. So if you put it 1024 units above the ground, for example, what would happen?

**Comment by Luke18033** [comment]
Ah, it's on cables. Never mind.

**Comment by libbybapa** [comment]
What is the 'Double Grated Panel'?

**Comment by TeamSpen210** [comment]
A bunch of 90-degree panels that unfold one after another in a line.

**Comment by Luke18033** [comment]
I think Bombs should actually be a separate item with the gel class. This would allow placing the dropper on walls. As noted above, dropperless bombs wouldn't generate. (Would it be possible to just force the dropper to be present?) 

The other cubes would be combined into one item using Cube Type:
- Standard = Bumbleball
- Companion = Sphere Cannon
- Reflection = Catapult (to get the laser preview)
- Sphere/Frankenturret = "Error" cube, reserved for future custom cubes.

**Comment by TeamSpen210** [comment]
I don't know if I can even have custom of those, since the cube item isn't generated.

**Comment by Luke18033** [comment]
I noticed that you were able to make the turret dropper auto-respawn if the turret is fizzled, which you originally said you couldn't do. Why couldn't you do it before, and what did you do to make it work now?

**Comment by TeamSpen210** [comment]
Turrets don't have outputs to tell you when they are fizzled or killed by goo. The dropper now uses a script to do the functionallity. In that I keep a reference to the turret, and check every second or so if the turret no longer exists. I couldn't do it before because I couldn't be sure the original BEE2 packed in things (all the items were designed to work without that).

**Comment by Luke18033** [comment]
Oh, I know what to do for the cubes! Only make dropperless variants, but give them connections. When connected to the vanilla dropper, the cube would be deleted and the dropper would be switched to that cube type. Kind of like the standing fizzlers.

**Comment by TeamSpen210** [comment]
See #1674 for Worldportal Doors.

**Comment by Endermage77** [comment]
Is there a way to see what was edited in comments?

This thing was updated 12 minutes ago and I have no idea what's different about it.

**Comment by TeamSpen210** [comment]
Unfortunately no. It might be the mention in #1777.

**Comment by TeamSpen210** [comment]
It's possibly @Luke18033's comment.

**Comment by ZombiePwner1** [comment]
@Endermage77 I updated the post before this morning i ticked off the recessed fizzlers since they can be done through partial blocks.

**Comment by awesomedoctorx3** [comment]
I found this for the Portalable Platform, however I followed the instructions and it seems to not work:
http://forums.steampowered.com/forums/showthread.php?p=33964311#post33964311

**Comment by TeamSpen210** [comment]
It's for the old BEE2.

**Comment by IceTudor** [comment]
Skybox wall and Skybox item. Seriously, we need them.

**Comment by Endermage77** [comment]
Wow, it's incredible to think that it's been over a year since the master checklist was last updated.

**Comment by Endermage77** [comment]
I wonder why the old Skybox Tile isn't on here yet?

**Comment by King0191** [comment]
Considering the Toggle option for the Lifeform Sensor is already complete (and perhaps the timed if the Fizzler Output Relay's timer works with it) Shouldn't it be checked off?

**Comment by ZipZoptheCat** [comment]
I see someone mentioned angled and offset Hard light bridges but what about;
2x2 centered Hard Light Bridge
Angled Excursion Funnels
Angled, 2x2 Centered and offset Laser Emitter/Catcher
Angled and 2x2 Centered HEP Emitter/Catcher

Wall Pistons that are at an angle when extended but flush when retracted. Just like in the introduction to momentum map in P1. I've seen this mentioned elsewhere but not here.

Light strips you can change colour with the selector like you can with the square lights. They were like this in 2.2 plus there was a cool purple colour available.

Items that make a hole in the wall to the skybox. I think that's what Ice Tudor meant? Maybe have this replace observation rooms in old Aperture? An Observation room that is 2x1 is actually 4x3 can it make a 4x3 hole in the wall or is that not possible?

**Comment by Endermage77** [comment]
Sorry mate, this thread is **not** for making suggestions

This thread is a to-do list of implementing stuff that was in previous versions but haven't gotten back in yet

**Comment by Endermage77** [comment]
The edited function doesn't really say what's been changed about the tick box lists. What has changed?

**Comment by TeamSpen210** [comment]
Marked TNT Turrets as not being implemented.

**Comment by Konclan** [comment]
Probably for the best

**Comment by nar3ik** [comment]
so just a question, is this still being developed or not? i kind of want to know because i like making portal 2 maps and this makes it better.

**Comment by TeamSpen210** [comment]
What exactly do you mean, the mod, these items? The VSD is a door that splits in half up and down. It’s seen in many styles.

**Comment by Luke18033** [comment]
I thought of a way how we could animate Catapult Cubes, while also allowing them to show gel - `point_posecontroller`s. Those work on physics props (including cubes), so we can edit the model to be able to control the catapult with a pose parameter, then animate it manually in-game.

**Comment by Luke18033** [comment]
Though one thing I should point out is that posecontrollers will crash the game if the specified parameter doesn't exist, so if we use the script method to change the model, we'll want to make sure the posecontroller isn't spawned in until afterward.

**Comment by TeamSpen210** [comment]
Good idea! I don't think that'll be a problem actually - looking at the code it seems to wait 1 second before scanning for the props.

**Comment by Luke18033** [comment]
Oh, another thing about that entity - the Prop Name keyvalue is a `string` instead of a `target_destination` in the FGD, causing it to not recieve instance fixups. That's fixed in your addons, but not everyone will necessarily have those, so we'll need to use a comp_kv_setter to add the fixup manually.

**Comment by Luke18033** [comment]
What's the point of the Override, can't you replicate that setup with existing logic gate items?

**Comment by somedoodest** [comment]
Question: will the Clean/Overgrown variant of the Vertical Security Door use the Portal 2 model or the Portal 1 model? I would prefer Clean/Overgrown's variant using the Portal 1 model, as the Portal 2 model fits more for BTS.

**Comment by Luke18033** [comment]
We could maybe use the P1 model with P2 sounds, like at the entrance to GLaDOS' chamber in the campaign. Probably want to add a dirty skin as well.

**Comment by Luke18033** [comment]
I still think Bomb Droppers should be able to be placed like gel droppers. The main problem with that is the fact that adding a custom gel splat item causes the vanilla dropper to generate...but what if we add a custom *gel dropper* item instead, which causes the vanilla *splat* to generate? That wouldn't matter since it doesn't actually appear in-game, and solves the problem of dropperless bombs (since disabling the dropper would just convert it into a normal gel splat). Note that setting Gel Flow to Bomb on wall-mounted droppers causes a `trigger_catapult` to generate, which could be used for this.

**Comment by Luke18033** [comment]
If that still won't work, it could alternatively be a faith plate-type item. The target texture would be removed in-game, or possibly replaced by some sort of scorch mark. If the target was placed in an invalid location (to the sides or behind the dropper), it would be snapped to the nearest valid one in-game.

**Comment by TeamSpen210** [comment]
I added a faithplate system during the tiling update, which treats bomb-type droppers as a special kind of plate. Making it one of those sounds like a good idea.

**Comment by GiovanH** [comment]
Are ceiling lights still relevant, or can style-specific ceiling observation rooms (often just light sources) replace them?

**Comment by slambo312** [comment]
about what percentage of this stuff is done so far?

**Comment by Luke18033** [comment]
The things which are checked off?

**Comment by Luke18033** [comment]
It literally shows how much of the issue is done at the top of the page, unless that's a Refined GitHub feature.
![image](https://user-images.githubusercontent.com/19228201/131438339-32463cf6-d373-4e8d-a6dd-e66516ccb305.png)

**Comment by slambo312** [comment]
oh i'm dumb

**Comment by slambo312** [comment]
Also i kinda want the living wall to be ported. It would be useful for stuff like those tests _just_ after GladOS is woken up.

**Comment by slambo311** [comment]
All 3 checklists haven't been updated in months. Are they still up-to-date with what's pretty much done in the development versions?

**Comment by Luke18033** [comment]
Yeah, we haven't been working on this lately.

**Comment by TPEcool** [comment]
what exactly is the refurbished style?

**Comment by FrozenDragon0** [comment]
> what exactly is the refurbished style?

It's basically Old Aperture chambers but taken over by GLaDOS with modern test elements, it's similar to coop course 5 in the coop campaign.

**Comment by TPEcool** [comment]
Oh, ok, that would be pretty cool!

**Comment by Pixcore72** [comment]
Can we finally get breakable glass I feel like it has been left behind at the top of the forum.

**Comment by Konclan** [comment]
This isn't a forum, its a checklist. It'll be added eventually, please read the contributing guidelines and FAQ.

**Comment by LincoCompany00** [comment]
Or a suggestion for the signage and structure, mostly. It would receive an input that rotates it, and there'd be an arrow pointing in each direction it can work. Maybe it can accept multiple inputs, and the model's rotation would snap to 45° angles?

![Image](https://github.com/user-attachments/assets/6eac61e5-25ab-4b23-abbd-43a4e8935dca)

**Comment by vrad-exe** [comment]
I believe there is an unfinished version of the rotating faith plate on another branch and it does have a circular design like that. The signage is a neat idea, maybe we should make custom diagonal arrow signs like you showed in your image. I wonder if it would be possible to configure connection points on the item so that the antlines always connect to the side it's going to rotate towards or something.

**Comment by vrad-exe** [comment]
Your physical models of the items are really cool and unique btw, never seen someone make item concepts that way before

**Comment by TeamSpen210** [comment]
They are very nice indeed. I do like the idea of adding signage to indicate the possible positions, we could definitely do that. Antlines probably not. For reference, here's the design I came up with quite a long time ago:

![rotating faith plate](https://github.com/user-attachments/assets/3a24e73d-4bad-4380-a161-8d75a7573f97)

**Comment by LincoCompany00** [comment]
Thank you! I like to make test chambers on my desk, so that's what those handmade ones are for. They also work well for this, too!

**Comment by LincoCompany00** [comment]
You could probably make the signage link to a specific side unless that spot is obstructed, which case it would go to a different side like any other item. The Exit Door is like this, right?

If you can't link the antlines to a specific side, someone could use antline corners and/or logic gates to do it; I've learned to use those items to manipulate antlines as good as you'd be able to in hammer.

This picture shows the model in the last comment and another arrow/antline signage idea I have for if you are able to do the antlines/connections.

![Image](https://github.com/user-attachments/assets/9a60ca27-5cd5-4a13-bfe5-0c1b4b7ea481)

**Comment by LincoCompany00** [comment]
The diagonal arrow signage could probably be added to the signage item.

**Comment by TeamSpen210** [comment]
The problem with antlines is that I can only have a single fixed definition, it'd be tricky to have it always connect to the side you specify with the item - though actually maybe I can since the additional targets are their own items. Definitely would be neat to illuminate the arrow to indicate the destination.

**Comment by LincoCompany00** [comment]
The laser music should have an option to change it, like the funnel and gel music so you can use base music that doesn't have a laser music track, that way you can use a set of laser tracks that sound similar.

**Comment by LincoCompany00** [comment]
For the rotating faith plate, would the item you place on top basically be another faith plate, with its target, and you connect inputs to that item? That way you can have multiple directions! The original faith plate would be the position it defaults to when none of the other target items are activated. Inputs to the original faith plate toggle it's on/off state.

**Comment by valerakianskij-hub** [comment]
What's up with the TNT Turret? Is it cut out?

**Comment by vrad-exe** [comment]
It didn't really seem very useful so we're not planning to reimplement it unless maybe if there's significant demand for it (there hasn't been).
